hack/run-functests.sh

```diff
@@ -32,103 +41,86 @@ elif [ "$FEATURES" == "" ]; then
 	echo "No FEATURES provided"
   exit 1
 else
-  FOCUS="-ginkgo.focus="$(echo "$FEATURES" | tr ' ' '|')
+  FOCUS="--focus="$(echo "$FEATURES" | tr ' ' '|')
   if [ "$FOCUS_TESTS" != "" ]; then
-    FOCUS="-ginkgo.focus="$(echo "$FOCUS_TESTS" | tr ' ' '|')
+    FOCUS="--focus="$(echo "$FOCUS_TESTS" | tr ' ' '|')
   fi
   echo "Focusing on $FOCUS"
 fi
 
 if [ "$SKIP_TESTS" != "" ]; then
-	SKIP="-ginkgo.skip="$(echo "$SKIP_TESTS" | tr ' ' '|')
+	SKIP="--skip="$(echo "$SKIP_TESTS" | tr ' ' '|')
 	echo "Skip set, skipping $SKIP"
 fi
 
-export SUITES_PATH=cnf-tests/bin
-
-mkdir -p "$TESTS_REPORTS_PATH"
-
+go build -o cnf-tests/bin/junit-merger cnf-tests/testsuites/pkg/junit-merger/junit-merger.go
 
+TEST_SUITES=${TEST_SUITES:-"validationsuite configsuite cnftests"}
+suites=( $TEST_SUITES )
+REPORTERS=""  
 
-if [ "$TESTS_IN_CONTAINER" == "true" ]; then
-  cp -f "$KUBECONFIG" _cache/kubeconfig
-  echo "Running dockerized version via $TEST_EXECUTION_IMAGE"
-
-  features="$(echo "$FEATURES" | tr ' ' '|')"
-
-  env_vars="-e CLEAN_PERFORMANCE_PROFILE=false \
-  -e CNF_TESTS_IMAGE=$TEST_POD_CNF_TEST_IMAGE \
-  -e DPDK_TESTS_IMAGE=$TEST_POD_DPDK_TEST_IMAGE \
-  -e IMAGE_REGISTRY=$TEST_POD_IMAGES_REGISTRY \
-  -e KUBECONFIG=/kubeconfig/kubeconfig \
-  -e SCTPTEST_HAS_NON_CNF_WORKERS=$SCTPTEST_HAS_NON_CNF_WORKERS \
-  -e TEST_SUITES=$TEST_SUITES \
-  -e IS_OPENSHIFT=$IS_OPENSHIFT \
-  -e FEATURES=$features \
-  -e OO_INSTALL_NAMESPACE=$OO_INSTALL_NAMESPACE"
-
-  # add latency tests env variable to the cnf-tests container
-  if [ "$LATENCY_TEST_RUN" == "true" ];then
-    env_vars="$env_vars \
-    -e LATENCY_TEST_RUN=$LATENCY_TEST_RUN \
-    -e LATENCY_TEST_RUNTIME=$LATENCY_TEST_RUNTIME \
-    -e LATENCY_TEST_DELAY=$LATENCY_TEST_DELAY \
-    -e OSLAT_MAXIMUM_LATENCY=$OSLAT_MAXIMUM_LATENCY \
-    -e CYCLICTEST_MAXIMUM_LATENCY=$CYCLICTEST_MAXIMUM_LATENCY \
-    -e HWLATDETECT_MAXIMUM_LATENCY=$HWLATDETECT_MAXIMUM_LATENCY \
-    -e MAXIMUM_LATENCY=$MAXIMUM_LATENCY"
+for suite in "${suites[@]}"; do
```

Comment:
We are mixing the `suite` term for both the set of tests (metallb, sriov) and for the step (validation, config, e2etest)
I'd suggest renaming the "suite" to "validationstep" or just step or something like that.

Reply:
I renamed to "step" but note that I left:
```
TEST_SUITES=${TEST_SUITES:-"validationsuite configsuite cnftests"}
steps=( $TEST_SUITES )
```
So that the API will remain the same for anyone who uses this env var from outside.

Reply:
regardless, I think it's much clearer now.